tunnel/intercept/tproxy/tproxy_linux.go

```diff
@@ -467,35 +493,62 @@ func (self *tProxy) addInterceptAddr(interceptAddr *intercept.InterceptAddress,
 	tracker.AddAddress(ipNet.String())
 	self.addresses = append(self.addresses, interceptAddr)
 
-	interceptAddr.TproxySpec = []string{
-		"-m", "comment", "--comment", service.Name,
-		"-d", ipNet.String(),
-		"-p", interceptAddr.Proto(),
-		"--dport", fmt.Sprintf("%v:%v", interceptAddr.LowPort(), interceptAddr.HighPort()),
-		"-j", "TPROXY",
-		"--tproxy-mark", "0x1/0x1",
-		fmt.Sprintf("--on-ip=%s", port.GetIP().String()),
-		fmt.Sprintf("--on-port=%d", port.GetPort()),
-	}
+	var tproxyAddr netip.AddrPort
 
-	pfxlog.Logger().Infof("Adding rule iptables -t %v -A %v %v", mangleTable, dstChain, interceptAddr.TproxySpec)
-	if err := self.interceptor.ipt.Insert(mangleTable, dstChain, 1, interceptAddr.TproxySpec...); err != nil {
-		return errors.Wrap(err, "failed to insert rule")
-	}
-
-	if self.interceptor.lanIf != "" {
-		interceptAddr.AcceptSpec = []string{
-			"-i", self.interceptor.lanIf,
+	switch interceptAddr.Proto() {
+	case "tcp":
+		tproxyAddr = self.tcpLn.Addr().(*net.TCPAddr).AddrPort()
+	case "udp":
+		tproxyAddr = self.udpLn.LocalAddr().(*net.UDPAddr).AddrPort()
+	default:
+		return errors.Errorf("unknown protocol %s for service %s", interceptAddr.Proto(), service.Name)
+	}
+
+	if self.interceptor.diverter != "" {
+		cidr := strings.Split(ipNet.String(), "/")
+		cmd := exec.Command(self.interceptor.diverter, "-I",
```

Comment:
cidr[0] and cidr[1] are directly referenced but I don't remember seeing any validation that cidr won't be nil and contains a slash in it. should you add some validation here to prevent a possible panic on user error?

Reply:
I don't think it could happen, but checking seems wise and easy.